### PR TITLE
ktor 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by Stexxe**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/k/ktor.rb
Comment (carlocab) [inline Formula/k/ktor.rb:6]: Do we need a Git clone here? Can we use a tarball?

File: Formula/k/ktor.rb
Comment (carlocab) [inline Formula/k/ktor.rb:15]:
```suggestion
    ldflags = "-X main.Version=#{version}"
    system "go", "build", *std_go_args(ldflags:), "github.com/ktorio/ktor-cli/cmd/ktor"
```

File: Formula/k/ktor.rb
Comment (carlocab) [inline Formula/k/ktor.rb:20]:
```suggestion
    assert_path_exists testpath/"project/build.gradle.kts"
```
and similarly below.

**Review by carlocab — COMMENTED** [review]
> There is one audit error: `GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)`. Can you please make an exception since the publisher is JetBrains?

Do you have any other indicator for how widely used this project is?

File: Formula/k/ktor.rb
Comment (Stexxe) [reply Formula/k/ktor.rb:6]: What's the problem with git clone?

Comment (carlocab) [reply Formula/k/ktor.rb:6]: > What's the problem with git clone?

Git clones are slower and worse for caching, so we should avoid it unless we need it.

> Do you mean to use an URL like `https://github.com/ktorio/ktor-cli/archive/0.2.0.tar.gz` and `sha256`?

Yes, exactly.

**Comment by Stexxe** [comment]
> > There is one audit error: `GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)`. Can you please make an exception since the publisher is JetBrains?
> 
> Do you have any other indicator for how widely used this project is?

Should all requirements be met (forks, watchers, stars), or the number of stars is enough?

**Comment by carlocab** [comment]
> > > There is one audit error: `GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)`. Can you please make an exception since the publisher is JetBrains?
> > 
> > 
> > Do you have any other indicator for how widely used this project is?
> 
> Should all requirements be met (forks, watchers, stars), or the number of stars is enough?

IIRC you only need to meet one of {forks, watchers, stars} to satisfy the audit.

**Comment by Stexxe** [comment]
> > > > There is one audit error: `GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)`. Can you please make an exception since the publisher is JetBrains?
> > > 
> > > 
> > > Do you have any other indicator for how widely used this project is?
> > 
> > 
> > Should all requirements be met (forks, watchers, stars), or the number of stars is enough?
> 
> IIRC you only need to meet one of {forks, watchers, stars} to satisfy the audit.

Now, the repository has enough stars to satisfy the audit.

File: Formula/k/ktor.rb
Comment (carlocab) [inline Formula/k/ktor.rb:18]: Let's check that the `-X main.Version` flag we pass above is correct with something like:
```suggestion
  test do
    assert_match version.to_s, shell_output("#{bin}/ktor version")
    system bin/"ktor", "new", "project"
```

**Review by carlocab — DISMISSED** [review]
LGTM, but I'll wait till another maintainer has had the chance to review this.

**Review by nandahkrishna — DISMISSED** [review]
Looks good to me as well.

**Pull request merged**
